diff --git a/init.lua b/init.lua
index 2a44edf..d834d97 100644
--- a/init.lua
+++ b/init.lua
@@ -1866,7 +1866,7 @@ function lsp.request_document_format(doc)
           options = {
             tabSize = config.indent_size,
             insertSpaces = config.tab_type == "soft",
-            trimTrailingWhitespace = config.plugins.trimwhitespace or true,
+            trimTrailingWhitespace = config.plugins.trimwhitespace.enabled or true,
             insertFinalNewline = false,
             trimFinalNewlines = true
           }
